lib/tasks/assets.rake

```diff
@@ -1,18 +1,14 @@
 namespace :assets do
-  desc "Mark an asset as deleted"
-  task :delete, [:id] => :environment do |_t, args|
-    Asset.find(args.fetch(:id)).destroy
-  end
-
-  desc "Restore an asset after being deleted"
-  task :restore, [:id] => :environment do |_t, args|
-    Asset.find(args.fetch(:id)).restore
-  end
-
   desc "Mark an asset as deleted and remove from S3"
-  task :delete_and_remove_from_s3, [:id] => :environment do |_t, args|
+  task :delete, %i[id permanent] => :environment do |_t, args|
     asset = Asset.find(args.fetch(:id))
     asset.destroy
-    Services.cloud_storage.delete(asset)
+    Services.cloud_storage.delete(asset) if permanent
```

Comment:
```suggestion
    Services.cloud_storage.delete(asset) if args[:permanent]
```

Although I'm not sure if this works for booleans, in case someone passes in `"false"` which would evaluate to true...

Reply:
Oops - fixed.